### PR TITLE
[#319] Added possibility to change default operation timeout.

### PR DESCRIPTION
**Pull request by uKL**
[#319] Added possibility to change default operation timeout.

### CODE REVIEW

File: rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java [hunk 1]
```diff
@@ -87,6 +87,24 @@
...
+     *                                         was already established or was in pending connection state. This occurs when the
+     *                                         connection was released as a part of expected behavior
+     *                                         (with {@link android.bluetooth.BluetoothGatt#GATT_SUCCESS} state).
+     * @throws BleGattException                emitted when the BLE link has been interrupted as a result of an error.
+     *                                         The exception contains detailed explanation of the error source (type of operation) and
+     *                                         the code proxied from the Android system.
+     * @throws BleGattCallbackTimeoutException emitted when an internal timeout for connection has been reached. The operation will
+     *                                         timeout in direct mode (autoConnect = false) after 35 seconds.
+     */
+    Observable<RxBleConnection> establishConnection(ConnectionSetup connectionSetup);
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java]: I wanted to wait with the signature `establishConnection(ConnectionSetup)` till the new `Connection` API will be ready
Reasoning:
* `establishConnection(ConnectionSetup)` is the easiest possible entry point
* if we will use it with now—we will not be able to introduce the new `Connection` API without coming up with a new name for the function

Maybe we could use something like `Observable<RxBleConnection> establishConnection(boolean autoConnect, TimeoutSetup timeoutSetup)` for now?

File: settings.gradle [hunk 2]
```diff
@@ -1,3 +1,4 @@
-include ':rxandroidble', ':mockrxandroidble', ':sample'
+include ':rxandroidble'
```
Comment (dariuszseweryn) [inline settings.gradle]: is this relevant?

Comment (uKL) [reply settings.gradle]: No, a bug. Will revert.

File: rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java (hunk 1 above)
Comment (uKL) [reply rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java]: It's ok for me.

**Comment by uKL** [comment]
@dariuszseweryn Ready to be checked again.

**Comment by uKL** [comment]
@dariuszseweryn Ready to be checked again.

File: rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java [hunk 3]
```diff
@@ -87,6 +87,42 @@
...
+     *                                         was already established or was in pending connection state. This occurs when the
+     *                                         connection was released as a part of expected behavior
+     *                                         (with {@link android.bluetooth.BluetoothGatt#GATT_SUCCESS} state).
+     * @throws BleGattException                emitted when the BLE link has been interrupted as a result of an error.
+     *                                         The exception contains detailed explanation of the error source (type of operation) and
+     *                                         the code proxied from the Android system.
+     * @throws BleGattCallbackTimeoutException emitted when an internal timeout for connection has been reached. The operation will
+     *                                         timeout in direct mode (autoConnect = false) after 35 seconds.
+     */
+    Observable<RxBleConnection> establishConnection(boolean autoConnect, TimeoutSetup operationTimeoutSetup);
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java]: Please add `@NonNull` annotation

File: rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+public class TimeoutSetup {
+
+    public final TimeUnit timeUnit;
+    public final long timeout;
+
+    public TimeoutSetup(long timeout, TimeUnit timeUnit) {
+        this.timeUnit = timeUnit;
+        this.timeout = timeout;
+    }
+}
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java]: new line

File: rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.polidea.rxandroidble;
+
+import java.util.concurrent.TimeUnit;
+
+public class TimeoutSetup {
+
+    public final TimeUnit timeUnit;
+    public final long timeout;
+
+    public TimeoutSetup(long timeout, TimeUnit timeUnit) {
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java]: Maybe just `Timeout`?

Comment (uKL) [reply rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java]: ok

File: rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java (hunk 4 above)
Comment (uKL) [reply rxandroidble/src/main/java/com/polidea/rxandroidble/TimeoutSetup.java]: ok

File: rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java (hunk 3 above)
Comment (uKL) [reply rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleDevice.java]: ok

**Pull request merged**